src/scenarios/billing/SubscribeToSubscriptionPlan/SubscribeToSubscriptionPlan.tsx

```diff
@@ -0,0 +1,476 @@
+          Subscribe to a Subscription Plan (with Paypal purchase)
+        </Header>
+        <Header size="small">
+          Required Services:
+          <Label>user-service</Label>
+          <Label>billing-service</Label>
+        </Header>
+
+        <Divider />
+
+        <Container fluid>
+          <p>
+            This scenario demonstrates the purchase flow of a PayPal-based
+            payment plan for a logged-in user.
+          </p>
+          <p>
+            If the user is not already signed-in, you can use one of the Sign-In
+            scenarios.
+          </p>
+          <p>
+            Billing Service should be configured with the following settings for
+            this scenario to work:
+          </p>
+          <p>
+            <b>Success Redirect URL:</b> {window.location.href}
+            /subscribe-success
+          </p>
+          <p>
+            <b>Error Redirect URL:</b> {window.location.href}/subscribe-error
+          </p>
+        </Container>
+
+        <Divider />
+
+        <Form>
+          <Form.Input
+            control={VariableSearch}
+            width={4}
+            icon="key"
+            label="User Access Token"
+            value={userAccessToken}
+            setStateValue={setUserAccessToken}
+          />
+
+          <Button
+            primary
+            onClick={async () => {
+              fetchPaymentPlans();
+            }}
+          >
+            Fetch Payment Plans & Paypal Settings
+          </Button>
+
+          <Divider />
+
+          <Form.Dropdown
+            fluid
+            selection
+            width={4}
+            label="Subscription Plans with PayPal Purchase"
+            placeholder="Select a subscription plan"
+            options={paymentPlans.map((plan) => {
+              return {
+                text: `[${plan.subscriptionPlan.title}] ${plan.title} (${
+                  plan.periodQuantity
+                } ${plan.periodUnit.toLocaleLowerCase()})`,
+                value: plan.id,
+              };
+            })}
+            value={paymentPlanId}
+            onChange={async (event, { value }) => {
+              const paymentPlanId = value as string;
+              setPaymentPlanId(paymentPlanId);
+              await fetchPaymentPlanPrices(paymentPlanId);
+            }}
+          ></Form.Dropdown>
+
+          <Form.Dropdown
+            fluid
+            selection
+            width={4}
+            label="Country Prices"
+            placeholder="Select an option"
+            options={paymentPlanPrices.map((option) => {
+              return {
+                text: `${option.country} [${option.price} ${option.currency}]`,
+                value: option.country,
+              };
+            })}
+            value={country}
+            onChange={(event, { value }) => {
+              setCountry(value as string);
+            }}
+          ></Form.Dropdown>
+
+          <Form.Field label="Purchase Flow"></Form.Field>
```

Comment:
we could implement some logic here to keep these UI elements disabled from user interaction until there's some actionable value present in the above fields?